lib/internal/process/execution.js

```diff
@@ -60,35 +60,44 @@ function evalScript(name, body, breakFirstLine, print) {
   module.filename = path.join(cwd, name);
   module.paths = CJSModule._nodeModulePaths(cwd);
 
+  const { handleMainPromise } = require('internal/modules/run_main');
   const asyncESM = require('internal/process/esm_loader');
   const baseUrl = pathToFileURL(module.filename).href;
+  const { loadESM } = asyncESM;
+
+  const runScript = () => {
```

Comment:
I can't really speak to what's happening in this file. I see a few others have already approved, so I'm assuming it's 👍

Reply:
diff here is really bad, just took some code and extracted to a function